xarray/conventions.py

```diff
@@ -384,11 +384,18 @@ def _update_bounds_encoding(variables: T_Variables) -> None:
                     bounds_encoding.setdefault("calendar", encoding["calendar"])
 
 
+def _item_or_default(obj: Mapping | Any, key: Hashable, default: Any = None):
```

Comment:
Untested:
```suggestion
T = TypeVar("T")

def _item_or_default(obj: Mapping[Any, T] | T, key: Hashable, default: T = None) -> T:
```
Otherwise you lose all typing when you use that because it returns Any.

Reply:
Great. Added import for typing.TypeVar and committed